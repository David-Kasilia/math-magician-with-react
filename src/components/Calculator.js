import React from 'react';
import '../App.css';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor() {
    super();
    this.state = {
      total: '0',
      next: null,
      operation: null,
    };

    this.onClickHandler = this.onClickHandler.bind(this);
  }

  onClickHandler = (e) => {
    if (!e.target.name) return;

    const { next, total, operation } = calculate(this.state, e.target.name);

    if (next === null && total === null) {
      this.setState({ next, total: '0', operation });
    } else {
      this.setState({ next, total, operation });
    }
  };

  render() {
    const { next, total } = this.state;
    const button = [
      'AC',
      '+/-',
      '%',
      'รท',
      '7',
      '8',
      '9',
      'x',
      '4',
      '5',
      '6',
      '-',
      '1',
      '2',
      '3',
      '+',
      '0',
      '.',
      '=',
    ];

    return (
      <div className="calculator-grid">
        {next ? (
          <div className="operand">
            {next}
          </div>
        ) : (
          <div className="operand">
            {total}
          </div>
        )}

        { button.map((buttonSymbol) => (
          <button
            onClick={(e) => this.onClickHandler(e)}
            name={buttonSymbol}
            type="button"
            className={`btn ${buttonSymbol === '0' ? 'btn-zero' : ''} 
            ${buttonSymbol === '=' ? 'btn-equal' : ''}
            ${buttonSymbol === 'รท' ? 'btn-divide' : ''}
            ${buttonSymbol === 'x' ? 'btn-multiply' : ''}
            ${buttonSymbol === '-' ? 'btn-subtract' : ''}
            ${buttonSymbol === '+' ? 'btn-add' : ''}`}
            key={buttonSymbol}
          >
            {buttonSymbol}
          </button>
        ))}

      </div>
    );
  }
}

export default Calculator;
