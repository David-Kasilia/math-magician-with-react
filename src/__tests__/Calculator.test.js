import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

test('Unit Test Check For Calculator Component', () => {
  render(<Calculator />);
  const linkElement = screen.getByText(/Let us do some math/i);
  expect(linkElement).toBeInTheDocument();
});

describe('Unit Test For Calculator component', () => {
  it('Calculator Component renders correctly', () => {
    const tree = TestRenderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
