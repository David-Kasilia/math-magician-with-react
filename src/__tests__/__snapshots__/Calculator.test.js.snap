// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test For Calculator component Calculator Component renders correctly 1`] = `
<div
  className="calculator"
>
  <h2
    className="title2"
  >
    Let us do some math
  </h2>
  <div
    className="calculator-grid"
  >
    <div
      className="operand"
    >
      0
    </div>
    <button
      className="btn  
            
            
            
            
            "
      name="AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn  
            
            
            
            
            "
      name="+/-"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn  
            
            
            
            
            "
      name="%"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn  
            
            btn-divide
            
            
            "
      name="รท"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="btn  
            
            
            
            
            "
      name="7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn  
            
            
            
            
            "
      name="8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn  
            
            
            
            
            "
      name="9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn  
            
            
            btn-multiply
            
            "
      name="x"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="btn  
            
            
            
            
            "
      name="4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn  
            
            
            
            
            "
      name="5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn  
            
            
            
            
            "
      name="6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn  
            
            
            
            btn-subtract
            "
      name="-"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="btn  
            
            
            
            
            "
      name="1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn  
            
            
            
            
            "
      name="2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn  
            
            
            
            
            "
      name="3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn  
            
            
            
            
            btn-add"
      name="+"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="btn btn-zero 
            
            
            
            
            "
      name="0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn  
            
            
            
            
            "
      name="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn  
            btn-equal
            
            
            
            "
      name="="
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
